### PR TITLE
Feature/reduce tests dependency

### PR DESCRIPTION
**Pull request by AnushkaRawatZ**
- Tried to fix the CI Pipeline.
- Removed duplicate dependencies from pom.xml
- Added extent reports to .gitignore file
- Created a function to navigate to the home page of Tamimi
- Refactored tests to utilize the said function instead of ```driver.navigate.back()```
- Removed the dependsOn attribute on search test and made it independent since it does not require successful execution of any other test to execute.

### CODE REVIEW

File: src/main/java/com/zs/pages/common/Flows.java [hunk 1]
```diff
@@ -44,24 +46,31 @@ public Flows(AndroidDriver driver, WebDriverWait wait){
     public void loginFlow(String password, String username, String appName){
         HomePage homePage=new HomePage(driver, wait);
         LoginPage loginPage=new LoginPage(driver, wait);
+
+        LoggerUtil.logInfo("Starting login flow for app: " + appName);
         // Switch based on the app name and execute the specific login steps.
-        switch(appName){
+        switch("Tamimi"){
```
Comment (sujeeth-kumar-zop) [inline src/main/java/com/zs/pages/common/Flows.java:52]: Is there a specific reason for providing Tamini app ?

File: src/main/java/com/zs/pages/common/HomePage.java [hunk 2]
```diff
@@ -59,9 +62,16 @@ public boolean isTextVisible(String text){
     }
 
     public String fetchRandomProduct(String appName){
+        LoggerUtil.logInfo("Fetching a random product to assert search.");
         List<WebElement> productsList = wait.until(ExpectedConditions.presenceOfAllElementsLocatedBy(CommonUtils.getHomePageLocator(appName,"productsInSearch")));
         int productCount = productsList.size();
-        int randomNumber= ThreadLocalRandom.current().nextInt(2, productCount);
+        int randomNumber= ThreadLocalRandom.current().nextInt(0, productCount);
```
Comment (sujeeth-kumar-zop) [inline src/main/java/com/zs/pages/common/HomePage.java:68]: Can this be moved to utils if it is being reused?

File: src/main/java/com/zs/utils/CommonUtils.java [hunk 3]
```diff
@@ -183,6 +185,37 @@ public boolean isElementVisible(By locator){
...
+    public void navigateToHome(String appName) {
+
+        int maxRetries = 10;
+        int attempts = 0;
+        WebDriverWait shortWait = new WebDriverWait(driver, Duration.ofSeconds(2));
+        while (attempts < maxRetries) {
+            try {
+
+                WebElement searchBar = shortWait.until(ExpectedConditions.visibilityOfElementLocated(CommonUtils.getHomePageLocator(appName, "searchPlaceholder")));
+                System.out.println(searchBar.isDisplayed());
```
Comment (sujeeth-kumar-zop) [inline src/main/java/com/zs/utils/CommonUtils.java:197]: remove sout statements, use logger instead.

File: src/main/java/com/zs/pages/common/HomePage.java (hunk 2 above)
Comment (AnushkaRawatZ) [reply src/main/java/com/zs/pages/common/HomePage.java:68]: It's a standard function provided by java to generate random numbers and is not being used anywhere else atleast right now. Should I still move it to utils?

File: src/main/java/com/zs/utils/CommonUtils.java (hunk 3 above)
Comment (AnushkaRawatZ) [reply src/main/java/com/zs/utils/CommonUtils.java]: Was using it for debugging. Have fixed it in the latest commit.

File: src/main/java/com/zs/pages/common/Flows.java (hunk 1 above)
Comment (AnushkaRawatZ) [reply src/main/java/com/zs/pages/common/Flows.java]: Did that to try understanding the ci pipeline. I've fixed it in the latest commit.

File: src/main/java/com/zs/pages/common/HomePage.java (hunk 2 above)
Comment (AnushkaRawatZ) [reply src/main/java/com/zs/pages/common/HomePage.java:68]: I don't think so

**Review by sujeeth-kumar-zop — APPROVED** [review]
Changes looks good.

**Pull request merged**